metadata-ingestion/src/datahub/ingestion/source/sql/hive.py

```diff
@@ -90,19 +96,39 @@ def dbapi_get_columns_patched(self, connection, table_name, schema=None, **kw):
     logger.warning(f"Failed to patch method due to {e}")
 
 
+try:
+    from pyhive.sqlalchemy_hive import HiveDialect
+
+    @reflection.cache  # type: ignore
+    def get_view_names_patched(self, connection, schema=None, **kw):
+        query = "SHOW VIEWS"
+        if schema:
+            query += " IN " + self.identifier_preparer.quote_identifier(schema)
+        return [row[0] for row in connection.execute(query)]
+
+    @reflection.cache  # type: ignore
+    def get_view_definition_patched(self, connection, view_name, schema=None, **kw):
+        full_table = self.identifier_preparer.quote_identifier(view_name)
+        if schema:
+            full_table = "{}.{}".format(
+                self.identifier_preparer.quote_identifier(schema),
+                self.identifier_preparer.quote_identifier(view_name),
+            )
+        row = connection.execute("SHOW CREATE TABLE {}".format(full_table)).fetchone()
+        return row[0]
+
+    HiveDialect.get_view_names = get_view_names_patched
+    HiveDialect.get_view_definition = get_view_definition_patched
+except ModuleNotFoundError:
+    pass
+except Exception as e:
```

Comment:
Failure to patch should cause the source to fail to load right?

Reply:
Right. Let me remove this exception handling.

Reply:
done